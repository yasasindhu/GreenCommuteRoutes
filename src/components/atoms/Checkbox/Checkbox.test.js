import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CheckboxWithGreenCheck from "./Checkbox";

describe("checkbox renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<CheckboxWithGreenCheck/>);
    expect(asFragment()).toMatchSnapshot();
  });

  test("checkbox can be clicked", () => {
    let checked = false;
    const handleChange = jest.fn();

    const { container } = render(
      <CheckboxWithGreenCheck
        checked={checked}
        onChange={handleChange}
        data-testid="checkBox"
      />
    );
    screen.getByTestId("checkBox");
    userEvent.click(container.querySelector("input"));
    expect(handleChange).toHaveBeenCalled();
  });
});
