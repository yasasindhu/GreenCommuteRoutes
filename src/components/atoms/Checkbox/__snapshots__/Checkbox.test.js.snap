// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root WithStyles(ForwardRef(Checkbox))-root-1 MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input-8"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</DocumentFragment>
`;
